src/graphql/graphql-api.php

```diff
@@ -0,0 +1,245 @@
+			}
+		}
+
+		return $flattened_blocks;
+	}
+
+	/**
+	 * Register types and fields graphql integration.
+	 *
+	 * @return void
+	 */
+	public static function register_types() {
+		// Register the type corresponding to the attributes of each individual block.
+		register_graphql_object_type(
+			'BlockDataAttribute',
+			[
+				'description' => 'Block data attribute',
+				'fields'      => [
+					'name'  => [
+						'type'        => 'String',
+						'description' => 'Block data attribute name',
+					],
+					'value' => [
+						'type'        => 'String',
+						'description' => 'Block data attribute value',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the individual inner block, with the above attribute.
+		register_graphql_type(
+			'InnerBlockData',
+			[
+				'description' => 'Block data',
+				'fields'      => [
+					'id'         => [
+						'type'        => 'String',
+						'description' => 'ID of the block',
+					],
+					'parentId'   => [
+						'type'        => 'String',
+						'description' => 'ID of the parent for this inner block, if it is an inner block. This will match the ID of the block',
+					],
+					'name'       => [
+						'type'        => 'String',
+						'description' => 'Block name',
+					],
+					'attributes' => [
+						'type'        => [
+							'list_of' => 'BlockDataAttribute',
+						],
+						'description' => 'Block data attributes',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the individual block, with the above attribute.
+		register_graphql_type(
+			'BlockData',
+			[
+				'description' => 'Block data',
+				'fields'      => [
+					'id'          => [
+						'type'        => 'String',
+						'description' => 'ID of the block',
+					],
+					'name'        => [
+						'type'        => 'String',
+						'description' => 'Block name',
+					],
+					'attributes'  => [
+						'type'        => [
+							'list_of' => 'BlockDataAttribute',
+						],
+						'description' => 'Block data attributes',
+					],
+					'innerBlocks' => [
+						'type'        => [ 'list_of' => 'InnerBlockData' ],
+						'description' => 'Flattened list of inner blocks of this block',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the list of individual blocks, with each item being the above type.
+		register_graphql_type(
+			'BlocksData',
+			[
+				'description' => 'Data for all the blocks',
+				'fields'      => [
+					'blocks'   => [
+						'type'        => [ 'list_of' => 'BlockData' ],
+						'description' => 'List of blocks data',
+					],
+					'warnings' => [
+						'type'        => [ 'list_of' => 'String' ],
+						'description' => 'List of warnings related to processing the blocks data',
+					],
+				],
+			],
+		);
+
+		// Register the field on every post type that supports 'editor'.
+		register_graphql_field(
+			'NodeWithContentEditor',
+			'BlocksData',
```

Comment:
field name should be camel case
```suggestion
			'blocksData',
```

Reply:
Changed this as well